instructions/section-1/lesson-1-4.md

```diff
@@ -0,0 +1,130 @@
+- Controlled Components
+- Props handling
+
+---
+
+## Instructions
+
+### Getting Started:
+
+Merge your pull request from the previous lesson (if you haven't already):
+
+[View tutorial](https://github.com/Code-the-Dream-School/common-instructions/blob/main/common/how-to-merge.md)
+
+Checkout your main branch and pull changes:
+
+    git checkout main
+    git pull
+
+Create a new local branch to work on separate from the `main` branch:
+
+    git checkout -b lesson-1-4
+
+Now, open the project directory in your code editor and continue to the next section.
+
+### Task List:
+
+#### Move Todo List into State
+
+- [ ] Open `/src/App.js`
+- [ ] Create new state variable named `todoList` with setter `setTodoList` and default value of an empty Array
+- [ ] Pass `todoList` state as a prop named `todoList` to the `TodoList` component
+- [ ] Open `/src/TodoList.js`
+- [ ] Add `props` as a parameter to the `TodoList` functional component
+- [ ] Change `todoList` to reference props instead of the hard-coded variable
+- [ ] Delete the hard-coded `todoList` variable
+- [ ] Run your application and view in browser
+    - [ ] Verify that your Todo List is now empty (no list items)
+
+#### Control "Add Todo" Input
+
+- [ ] Open `/src/AddTodoForm.js`
+- [ ] Create new state variable named `todoTitle` with setter `setTodoTitle`
+- [ ] Modify the `<input>` element to be a controlled input
+    - [ ] Add `value` prop equal to `todoTitle` from component props
+    - [ ] Add `onChange` prop equal to `handleTitleChange` function reference (we will declare this function in the next step)
+- [ ] Above the `handleAddTodo` function, declare a new function named `handleTitleChange` that takes `event` as a parameter
+    - [ ] First, retrieve the input value from the `event` object and store in variable named `newTodoTitle`
+    - [ ] Then, call the state setter `setTodoTitle` and pass `newTodoTitle`
+- [ ] In the `handleAddTodo` function, remove the `todoTitle` variable and update `onAddTodo` callback handler to pass our `todoTitle` state variable instead
+- [ ] Run your application and view in browser
+    - [ ] Enter a new todo in "Add Todo" form, submit, and verify that the title appears below
+
+#### Add New Todo to List
+
+- [ ] Open `/src/App.js`
+- [ ] Remove the `newTodo` state variable and the corresponding JSX that displays it
+- [ ] Declare a new function named `addTodo` that takes `newTodo` as a parameter
+    - [ ] Call the `setTodoList` state setter and use the spread operator to pass the existing Objects in the `todoList` Array along with the `newTodo` Object
+- [ ] Change the value of the `onAddTodo` prop for `AddTodoForm` to `addTodo`
+- [ ] Open `/src/AddTodoForm.js`
+- [ ] Inside `handleAddTodo`, update the `onAddTodo` callback prop to pass an Object instead of a String; Object should have the following properties:
+    - [ ] `title`: equal to `todoTitle`
+    - [ ] `id`: unique identifier (hint: use `Date.now()` to generate a unique number)
```

Comment:
This is handy, but I'm not sure how I feel about possibly setting the precedent of folks using a datetime as a unique ID in the future...

Reply:
do you have any other suggestions are generating a unique identifier? I went with this because it's simple and as long as todos can only be added one-at-a-time there won't be any conflicts

Reply:
Pull in a UUID library? Make a hash of some kind (based on the time)? Those sound kind of advanced for this stage, though. I see why you went with `Date.now()`, would just feel better if we had a disclaimer that this not something we'd do in production code.

Reply:
yeah I agree 100%, but wanted to avoid pulling in any libraries at this stage in their learning

I added a disclaimer for now